### PR TITLE
Add TLS configuration options

### LINKED ISSUE #885: Document TLS command line options
Confirm that that the dev issue has been merged.

See: https://github.com/hyperledger/besu/pull/3105

### PR DESCRIPTION
**Pull request by rolandtyler**
Signed-off-by: Roland Tyler <roland.tyler@consensys.net>

## Describe the change
Add TLS configuration options

## Issue fixed
Fixes #885 

### For content changes

- [x] Doc content
- [ ] Doc pages organisation

### For tools changes

- [ ] CircleCI workflow
- [ ] Build and QA tools (lint, vale,…)
- [ ] MkDocs templates
- [ ] MkDocs configuration
- [ ] Python dependencies
- [ ] Node dependencies and JavaScript
- [ ] ReadTheDocs configuration
- [ ] GitHub integration

## Testing
https://hyperledger-besu--900.org.readthedocs.build/en/900/Reference/CLI/CLI-Syntax/#rpc-http-tls-cipher-suite-rpc-http-tls-cipher-suites

https://hyperledger-besu--900.org.readthedocs.build/en/900/HowTo/Configure/TLS/Configure-TLS/#start-besu

### CODE REVIEW

**Review by macfarla — CHANGES_REQUESTED** [review]
config file version for list options need [] around them

File: docs/Reference/CLI/CLI-Syntax.md [hunk 1]
```diff
@@ -2997,6 +3025,34 @@ Must contain the certificates's Common Name, and SHA-256 fingerprint in the form
...
+=== "Environment variable"
+
+    ```bash
+    BESU_RPC_HTTP_TLS_PROTOCOL=TLSv1.3,TLSv1.2
+    ```
+
+=== "Configuration file"
+
+    ```bash
+    rpc-http-tls-protocol=["TLSv1.3,TLSv1.2"]
```
Comment (macfarla) [inline docs/Reference/CLI/CLI-Syntax.md:3051]: I think you want to have a list of strings. Did you try it?
    rpc-http-tls-protocol=["TLSv1.3","TLSv1.2"]

File: docs/Reference/CLI/CLI-Syntax.md [hunk 2]
```diff
@@ -2876,6 +2876,34 @@ Enables TLS client authentication for the JSON-RPC HTTP service. The default is
...
+=== "Environment variable"
+
+    ```bash
+    BESU_RPC_HTTP_TLS_CIPHER_SUITE=TLS_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256
+    ```
+
+=== "Configuration file"
+
+    ```bash
+    rpc-http-tls-cipher-suite=["TLS_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256"]
```
Comment (macfarla) [inline docs/Reference/CLI/CLI-Syntax.md:2902]: same here

Comment (macfarla) [reply docs/Reference/CLI/CLI-Syntax.md:2902]: rpc-http-tls-cipher-suite=["TLS_AES_256_GCM_SHA384","TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384","TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256"]

**Review by macfarla — APPROVED** [review]

File: docs/HowTo/Configure/TLS/Configure-TLS.md [hunk 3]
```diff
@@ -74,6 +74,10 @@ The command line:
   [`--rpc-http-tls-keystore-password-file`](../../../Reference/CLI/CLI-Syntax.md#rpc-http-tls-keystore-password-file) option.
 * [Specifies the clients](#create-the-known-clients-file) allowed to connect to Besu using the
   [`--rpc-http-tls-known-clients-file`](../../../Reference/CLI/CLI-Syntax.md#rpc-http-tls-known-clients-file) option.
+* specify the Java cipher suites using the
+  [`--rpc-http-tls-cipher-suite`](../../../Reference/CLI/CLI-Syntax.md#rpc-http-tls-cipher-suite-rpc-http-tls-cipher-suites) option.
+* specify the TLS protocol version using the
+  [`--rpc-http-tls-protocol`](../../../Reference/CLI/CLI-Syntax.md#rpc-http-tls-protocol-rpc-http-tls-protocols) option.
```
Comment (alexandratran) [inline docs/HowTo/Configure/TLS/Configure-TLS.md:80]:
```suggestion
* Specifies the Java cipher suites using the
  [`--rpc-http-tls-cipher-suite`](../../../Reference/CLI/CLI-Syntax.md#rpc-http-tls-cipher-suite-rpc-http-tls-cipher-suites) option.
* Specifies the TLS protocol version using the
  [`--rpc-http-tls-protocol`](../../../Reference/CLI/CLI-Syntax.md#rpc-http-tls-protocol-rpc-http-tls-protocols) option.
```

Note consistency with the rest of this list.

File: docs/Reference/CLI/CLI-Syntax.md [hunk 4]
```diff
@@ -2876,6 +2876,34 @@ Enables TLS client authentication for the JSON-RPC HTTP service. The default is
...
+    BESU_RPC_HTTP_TLS_CIPHER_SUITE=TLS_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256
+    ```
+
+=== "Configuration file"
+
+    ```bash
+    rpc-http-tls-cipher-suite=["TLS_AES_256_GCM_SHA384","TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384","TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256"]
+    ```
+
+A list of comma separated TLS cipher suites to support.
```
Comment (alexandratran) [inline docs/Reference/CLI/CLI-Syntax.md:2905]:
```suggestion
A list of comma-separated TLS cipher suites to support.
```

Maybe note here whether the singular and plural aliases can be used interchangeably, or a plural list requires the plural alias.

File: docs/Reference/CLI/CLI-Syntax.md [hunk 5]
```diff
@@ -2997,6 +3025,34 @@ Must contain the certificates's Common Name, and SHA-256 fingerprint in the form
...
+    BESU_RPC_HTTP_TLS_PROTOCOL=TLSv1.3,TLSv1.2
+    ```
+
+=== "Configuration file"
+
+    ```bash
+    rpc-http-tls-protocol=["TLSv1.3","TLSv1.2"]
+    ```
+
+A list of comma separated TLS protocols to support. Defaults to `DEFAULT_TLS_PROTOCOLS`, a list which includes `TLSv1.3` and `TLSv1.2`.
```
Comment (alexandratran) [inline docs/Reference/CLI/CLI-Syntax.md:3054]:
```suggestion
A list of comma-separated TLS protocols to support. The default is `DEFAULT_TLS_PROTOCOLS`, a list which includes `TLSv1.3` and `TLSv1.2`.
```

Same comment about singular vs. plural alias.

File: docs/Reference/CLI/CLI-Syntax.md [hunk 6]
```diff
@@ -2876,6 +2876,39 @@ Enables TLS client authentication for the JSON-RPC HTTP service. The default is
...
+    ```bash
+    rpc-http-tls-cipher-suite=["TLS_AES_256_GCM_SHA384","TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384","TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256"]
+    ```
+
+A list of comma-separated TLS cipher suites to support.
+
+!!!tip
+
+    The singular `--rpc-http-tls-cipher-suite` and plural `--rpc-http-tls-cipher-suites` are available and are two names for
+    the same option.
```
Comment (alexandratran) [inline docs/Reference/CLI/CLI-Syntax.md:2910]:
```suggestion
!!! note

    You can use `--rpc-http-tls-cipher-suite` and `--rpc-http-tls-cipher-suites` interchangeably.
```

File: docs/Reference/CLI/CLI-Syntax.md [hunk 7]
```diff
@@ -2997,6 +3030,39 @@ Must contain the certificates's Common Name, and SHA-256 fingerprint in the form
...
+    ```bash
+    rpc-http-tls-protocol=["TLSv1.3","TLSv1.2"]
+    ```
+
+A list of comma-separated TLS protocols to support. The default is `DEFAULT_TLS_PROTOCOLS`, a list which includes `TLSv1.3` and `TLSv1.2`.
+
+!!!tip
+
+    The singular `--rpc-http-tls-protocol` and plural `--rpc-http-tls-protocols` are available and are two names for
+    the same option.
```
Comment (alexandratran) [inline docs/Reference/CLI/CLI-Syntax.md:3064]:
```suggestion
!!! note

    You can use `--rpc-http-tls-protocol` and `--rpc-http-tls-protocols` interchangeably.
```

**Review by alexandratran — APPROVED** [review]
Final suggestions

**Pull request merged**